webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -1097,7 +1100,10 @@ protected function validateAndUpdateTeam(Event $event, EventData $data): void
         if (!empty($data->organizationId)) {
             $affiliation = $this->em->getRepository(TeamAffiliation::class)->findOneBy(['externalid' => $data->organizationId]);
             if (!$affiliation) {
+                // Affiliation does not exist. Create one with a dummy name so we can continue.
                 $affiliation = new TeamAffiliation();
+                $affiliation->setName($data->organizationId);
+                $affiliation->setShortname($data->organizationId);
```

Comment:
Can externalid be more chars than the shortname?

Reply:
It seems it can. We need to limit it to 32 characters so I will take the first 32.